Node selector is a dict

The-Loeki opened the pull request:
This PR is a 
 - [X] Feature Implementation
 - [X] Bug Fix

**What this PR does / why we need it**:
The `nodeSelector` parameters in the Helm chart are forced 'this: is not a string' 
This PR makes it dicts, improving usability and consistency

Please leave this checklist in the PR comment so that maintainers can ensure a good PR.

Merge Checklist:
 - [ ] Server Flag for config
   - [ ] Chart changes
   - [ ] removing a flag by marking deprecated and hiding to avoid
         breaking the chart release and existing clients who provide a
         flag that will get an error when they try to update

mszostok commented:
Hi @The-Loeki 

Thanks for your PR:)  please sign the CLA ;) https://github.com/kubernetes-sigs/service-catalog/pull/2733#issuecomment-543193820


/ok-to-test

mszostok commented on charts/catalog/templates/webhook-deployment.yaml:
```diff
@@ -83,7 +83,7 @@ spec:
       {{ if .Values.webhook.nodeSelector }}
       nodeSelector:
-          {{ .Values.webhook.nodeSelector }}
+{{ toYaml .Values.controllerManager.nodeSelector | indent(8) }}
```
```suggestion
{{ toYaml .Values.webhook.nodeSelector | indent(8) }}
```

mszostok commented:
Right now we have 3 jobs, they are short-lived and the `nodeSelector` was applied only for the main deployment of the Service Catalog: `controller-manager` and `webhook-server`. 

If you have a case that all Pods (even the short-lived e.g. from migration) related to the Service Catalog should be scheduled to the given node then we can specify in values.yaml:
```yaml
jobs:
  # nodeSelector to apply to pods from all Jobs which run to completion, e.g. cleaner, migration etc.
  nodeSelector:  
```

and add for each job:
```yaml
    {{ if .Values.controllerManager.nodeSelector }}
      nodeSelector:
{{ toYaml .Values.jobs.nodeSelector | indent 8 }}
    {{ end }}
```

mszostok commented:
tbh I do not get that, point:

> (..) there's a bunch of paths in the repo that are apparently too long...

what do you mean by that? you can execute just 
```
git clone git@github.com:kubernetes-sigs/service-catalog.git $GOPATH/src/github.com/kubernetes-sigs/service-catalog
```

and it should work

mszostok commented:
/test pull-build-all-images-for-arm

The-Loeki commented:
> you can execute just
> 
> ```
> git clone git@github.com:kubernetes-sigs/service-catalog.git $GOPATH/src/github.com/kubernetes-sigs/service-catalog
> ```
> 
> and it should work

```
[theloeki@murphy upstream]$ git clone git@github.com:kubernetes-sigs/service-catalog.git
Cloning into 'service-catalog'...
remote: Enumerating objects: 21, done.
remote: Counting objects: 100% (21/21), done.
remote: Compressing objects: 100% (21/21), done.
remote: Total 73652 (delta 5), reused 5 (delta 0), pack-reused 73631
Receiving objects: 100% (73652/73652), 106.18 MiB | 6.20 MiB/s, done.
Resolving deltas: 100% (37181/37181), done.
error: unable to create file cmd/svcat/testdata/responses/catalog/clusterserviceplans_labelSelector=servicecatalog.k8s.io_spec.clusterServiceClassRef.name=4f6e6cf6-ffdd-425f-a2c7-3c9258ad2468,servicecatalog.k8s.io_spec.externalName=default.json: File name too long
error: unable to create file cmd/svcat/testdata/responses/catalog/clusterserviceplans_labelSelector=servicecatalog.k8s.io_spec.externalName=default,servicecatalog.k8s.io_spec.serviceClassRef.name=4f6e6cf6-ffdd-425f-a2c7-3c9258ad2468.json: File name too long
error: unable to create file cmd/svcat/testdata/responses/catalog/namespaces/default/serviceplans_labelSelector=servicecatalog.k8s.io_spec.externalName=namespacedplan,servicecatalog.k8s.io_spec.serviceClassRef.name=4f6e6cf6-ffdd-425f-a2c7-3c9258ad2468.json: File name too long
error: unable to create file cmd/svcat/testdata/responses/catalog/namespaces/test-ns/serviceplans_labelSelector=servicecatalog.k8s.io_spec.clusterServiceClassRef.name=4f6e6cf6-ffdd-425f-a2c7-3c9258ad2468,servicecatalog.k8s.io_spec.externalName=default.jso: File name too long
error: unable to create file cmd/svcat/testdata/responses/catalog/namespaces/test-ns/serviceplans_labelSelector=servicecatalog.k8s.io_spec.externalName=default,servicecatalog.k8s.io_spec.serviceClassRef.name=4f6e6cf6-ffdd-425f-a2c7-3c9258ad2468.json: File name too long
Checking out files: 100% (4889/4889), done.
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry the checkout with 'git checkout -f HEAD'

[theloeki@murphy upstream]$ cd service-catalog/ ; git status
On branch master
Your branch is up to date with 'origin/master'.

Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        deleted:    .dockerignore
        deleted:    .github/ISSUE_TEMPLATE.md
        deleted:    .github/PULL_REQUEST_TEMPLATE.md
        deleted:    .gitignore
        deleted:    .travis.yml
       <<SNIP>>
Untracked files:
  (use "git add <file>..." to include in what will be committed)

        .dockerignore
        .github/
        .gitignore
        .travis.yml
        CONTRIBUTING.md
        Gopkg.lock
        Gopkg.toml
        Jenkinsfile
        LICENSE
        Makefile
        OWNERS
        README.md
        REVIEWING.md
        SECURITY_CONTACTS
        build/
        charts/
        cmd/
        code-of-conduct.md
        contrib/
        docs/
        docsite/
        internal/
        netlify.toml
        pkg/
        plugin/
        terminology.md
        test/
        vendor/
```

mszostok commented:
Hi @The-Loeki, any updates about this proposed solution: https://github.com/kubernetes-sigs/service-catalog/pull/2733#issuecomment-543223344

is that ok for you? Should I add missing things or you will do that?

The-Loeki commented:
Hi @mszostok sorry about that, this one fell a bit off my radar for now ;)

The fix didn't work (I'm on Fedora 30 btw, which somehow exhibits this behaviour without being Windows; I've Googled around a little bit and found mostly the same fix as you proposed)...

I did however roll out the SC a few more times this week, and still never ran into a dying job for not being on a Linux machine, so I'm  guessing it's not really necessary to add those annotations just yet, although I still can't think why; if they're not there, I'd expect the scheduler to be fully free to schedule a job just like any other app (and as the Windows node sits near-empty for now that would've been a prime candidate in any way), so I'm still a bit confused by that :)

The-Loeki commented:
```
[theloeki@murphy upstream]$ git clone https://github.com/kubernetes-sigs/service-catalog.git
Cloning into 'service-catalog'...
remote: Enumerating objects: 73652, done.
remote: Total 73652 (delta 0), reused 0 (delta 0), pack-reused 73652
Receiving objects: 100% (73652/73652), 106.18 MiB | 6.64 MiB/s, done.
Resolving deltas: 100% (37191/37191), done.
Checking out files: 100% (4889/4889), done.
```

Temporary insanity, that's what it was...

mszostok commented:
@The-Loeki I'm merging this pr because we will release new version today and is nice to have that fix. Thanks for contributing. If you will encounter any other problems then please submit another pr:)

Cheers!

/lgtm
/approve
